playbooks/roles/kibana/tasks/kibana_post_install.yml

```diff
@@ -0,0 +1,132 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+- name: create self-signed SSL cert
+  command: >
+    openssl req -new -nodes -x509 -subj
+    "/C=US/ST=Texas/L=San Antonio/O=IT/CN={{ kibana_server_name }}"
+    -days 365
+    -keyout /etc/ssl/private/apache.key
+    -out /etc/ssl/certs/apache.cert
+    -extensions v3_ca
+    creates=/etc/ssl/certs/apache.cert
+  when: kibana_self_signed is defined and kibana_self_signed == true
+  tags: kibana-self-signed-cert
+
+- name: Enable apache modules
+  command: a2enmod "{{ item }}"
+  with_items: kibana_apache_modules
+  tags:
+    - kibana-apache-modules
+    - kibana-apache
+
+- name: Template Kibana Apache Config
+  template:
+    src: "{{ item }}"
+    dest: "/etc/apache2/sites-available/{{ item }}"
+    owner: "{{ kibana_system_user }}"
+    group: "{{ kibana_system_group }}"
+  with_items:
+    - 000-kibana.conf
+  notify: Restart Apache
+  tags:
+    - kibana-apache
+    - kibana-config
+
+- name: Fix Apache2 Paths
+  template:
+    src: "{{ item }}"
+    dest: "/etc/apache2/{{ item }}"
+    owner: "root"
+    group: "root"
+  with_items:
+    - apache2.conf
+  notify: Restart Apache
+  tags:
+    - kibana-apache
+    - kibana-config
+
+- name: Drop Apache2 Ports File
+  template:
+    src: "{{ item }}"
+    dest: "/etc/apache2/{{ item }}"
+    owner: "root"
+    group: "root"
+  with_items:
+    - ports.conf
+  notify: Restart Apache
+  tags:
+    - kibana-apache
+    - kibana-config
+
+- name: Kibana Config
+  template:
+    src: "{{ item }}"
+    dest: "{{ kibana_root }}/{{ item }}"
+    owner: "{{ kibana_system_user }}"
+    group: "{{ kibana_system_group }}"
+  with_items:
+    - config.js
+  notify: Restart Apache
+  tags:
+    - kibana-config
+
+- name: Link Kibana Site
+  file:
+    state: "link"
+    src: "/etc/apache2/sites-available/000-kibana.conf"
+    dest: "/etc/apache2/sites-enabled/000-kibana.conf"
+    owner: "{{ kibana_system_user }}"
+    group: "{{ kibana_system_group }}"
+  notify: Restart Apache
+  tags:
+    - kibana-apache
+    - kibana-config
+
+- name: Remove Apache Default Site
+  file:
+    state: "absent"
+    path: "/etc/apache2/sites-enabled/000-default.conf"
+  notify: Restart Apache
+  tags:
+    - kibana-apache
+    - kibana-config
+
+- name: Create kibana http_auth user
+  htpasswd:
+    path: "/etc/apache2/users"
+    name: "kibana"
+    password: "{{ kibana_password }}"
+    owner: "root"
+    group: "www-data"
+    mode: "0640"
+    create: "yes"
+    state: "present"
+  notify: Restart Apache
+  tags:
```

Comment:
it's be nice to have the pre-install, install and post-install tags added in the same way as was done for the logstash/elasticsearch roles.